lib/mix/tasks/swagger.generate.ex

```diff
@@ -131,22 +131,19 @@ defmodule Mix.Tasks.Phoenix.Swagger.Generate do
     case function_exported?(Mix.Project.get, :swagger_info, 0) do
       true ->
         info = Mix.Project.get.swagger_info
-        title = Keyword.get(info, :title, nil)
-        version = Keyword.get(info, :version, nil)
+
+        # :title and :version are mandatory fields,
+        # so we need to add default values
+        # if they are not exists
+        title = Keyword.get(info, :title, @default_title)
+        version = Keyword.get(info, :version, @default_version)
+
         # collect :info swagger fields to the map
         info = List.foldl(info, %{},
           fn ({info_key, info_val}, map) ->
             Map.put_new(map, info_key, info_val)
           end)
-        # :title and :version are mandatory fields,
-        # so we need to check and add default values
-        # if they are not exists
-        info = if title == nil do
-          Map.put_new(info, :title, @default_title)
```

Comment:
I don't think this is correct at all. Removal this setter breaks the intended logic completely.